pages/api/dpu/repos.ts

```diff
@@ -1,25 +1,37 @@
 import { NextApiRequest, NextApiResponse } from "next";
-import { getUserRepositoriesByTopic } from "../../../utils/db/repos";
 import { RepoIdentifier } from "../../../types/repository";
+import { getUserRepositoriesByTopic } from "../../../utils/db/repos";
+import { getUserIdByTopicName } from "../../../utils/db/users";
+import rudderStackEvents from '../events';
 
 const reposHandler = async (req: NextApiRequest, res: NextApiResponse) => {
-    const { topicId, provider } = req.query;
-    if (!topicId || !provider 
-        || Array.isArray(topicId) || Array.isArray(provider)
-        || topicId.length === 0 || provider.length === 0) {
-        res.status(400).json({ error: 'Invalid get request body' });
+	const { topicId, provider } = req.query;
+	const userId = await getUserIdByTopicName(topicId as string).catch((error: any) => {
+		console.error("[setupHandler/getUserIdByTopicName] Failed to fetch userId from the database.", error);
+		return "";
+	});
+	if (!topicId || !provider 
+		|| Array.isArray(topicId) || Array.isArray(provider)
+		|| topicId.length === 0 || provider.length === 0) {
+		res.status(400).json({ error: 'Invalid get request body' });
+		const eventProperties = { ...req.query, response_status: 400 };
+		rudderStackEvents.track(userId, "", 'repos', { type: 'get-user-repos', eventStatusFlag: 0, eventProperties });
+		return;
+	}
+	const repoList: RepoIdentifier[] | null = await getUserRepositoriesByTopic(topicId, provider).catch((err) => {
+		console.error(`[reposHandler] Unable to get user repos for topic ${topicId}`);
+		return null;
+	});
+	if (repoList == null) {
+		res.status(500).json({"error": "Unable to get user repos from db"});
+		const eventProperties = { ...req.query, response_status: 500 };
+		rudderStackEvents.track(userId, "", 'repos', { type: 'get-user-repos', eventStatusFlag: 0, eventProperties });
 		return;
-    }
-    const repoList: RepoIdentifier[] | null = await getUserRepositoriesByTopic(topicId, provider).catch((err) => {
-        console.error(`[reposHandler] Unable to get user repos for topic ${topicId}`);
-        return null;
-    });
-    if (repoList == null) {
-        res.status(500).json({"error": "Unable to get user repos from db"});
-        return;
-    }
+	}
 	res.status(200).json({repoList: repoList});
-    return;
+	const eventProperties = { ...req.query, repoList, response_status: 200 };
```

Comment:
LOGAF: 5/10

Can you check how BigQuery stores array information? Does it do that elegantly (as an array-type or JSON-type column) or does it make a mess by generating many many columns? 
If it makes a mess, maybe we can just send some meta data here, like the length of the `repoList`.